experimental/PropertyDDS/packages/property-inspector-table/src/InspectorTable.tsx

```diff
@@ -262,715 +246,578 @@ class InspectorTable extends React.Component<WithStyles<typeof styles> & IInspec
-      // This has the undesired side effect that we also restart search when the browser window is resized, for example.
-      if ((prevProps !== this.props || prevState.sortBy.order !== sortBy.order) && !checkoutInProgress) {
-        if (data) {
-          const updatedTableRows = toTableRows({ data, id: "" } as IInspectorRow, this.props,
-            this.toTableRowOptions);
-          fillExpanded(expanded, updatedTableRows, this.props, this.toTableRowOptions);
-          // We need to update the table rows directly, because they might be used in the search call below.
-          // Treating table rows as a mutable state property is fine, since it is purely derived from props anyway, and
-          // we also update it directly in other places already.
-          this.state.tableRows = updatedTableRows;
-          // We still need to add it to the new state to trigger a re-render of the table.
-          newState.tableRows = updatedTableRows;
-        }
-
-        if (searchExpression.length > 0) {
-          // If a search process was running already, stop it.
-          if (this.state.searchAbortHandler) {
-            this.state.searchAbortHandler();
-          }
-
-          // Trigger the new search and update the state.
-          const searchControls = this.startSearch(searchExpression, this.updateSearchState, false);
-          newState.searchAbortHandler = searchControls ? searchControls.abortHandler : undefined;
-          newState.searchState = searchControls ? searchControls.state : undefined;
-          newState.searchDone = false;
-          newState.searchInProgress = true;
-          newState.currentResult = -1;
-          newState.foundMatches = [];
-          newState.matchesMap = {};
-          newState.childToParentMap = {};
-
-          foundMatches = [];
-          childToParentMap = {};
-          forceUpdateRequired = true;
-        }
-      }
+      // Set the initial state for a fresh search.
+      this.setState({ ...this.state, ...newState });
+    }, 250);
+  }
 
-      // Scroll the table to the current search result.
-      // TODO: As above, we should check for changes in props.data rather than just props.
-      let scrollingRequired = false;
-      let toExpand;
-      if (searchExpression.length > 0 && foundMatches.length > 0 && !checkoutInProgress &&
-          prevState.currentResult !== currentResult) {
-        // need to update expanded according to the new currentResult
-        toExpand = showNextResult(tableRows, expanded, foundMatches, currentResult!, childToParentMap);
-        newState.expanded = toExpand.expandedRows;
-        forceUpdateRequired = true;
-        scrollingRequired = true;
+  public componentDidMount() {
+    const { data, fillExpanded } = this.props;
+    const { expanded } = this.state;
+    if (data) {
+      const updatedTableRows = this.props.toTableRows!({ data, id: "" }, this.props, this.toTableRowOptions);
+      if (fillExpanded) {
```

Comment:
Logic changed. Before it would run `fillExpanded` in any case. Now, only if provided.